src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,168 @@
+            resizeList(key, value);
+            return;
+        }
+        int hashcode = hashCode(key);
+        Node<K,V> newNode = new Node<>(key, value);
+        if (key == null && nodes[0] == null) {
+            nodes[0] = newNode;
+            size++;
+            return;
+        } else if (key == null) {
+            nodes[0] = newNode;
+            return;
+        }
+        int index = defineIndex(hashcode);
+        Node<K,V> lastNode = getLastCollisionNode(index);
+        if (lastNode == null) {
+            nodes[index] = newNode;
+            size++;
+        } else {
+            reassignSameKeyValues(newNode, nodes[index]);
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        if (key == null) {
+            return nodes[0] != null ? nodes[0].getValue() : null;
+        }
+        int hashcode = hashCode(key);
+        int index = defineIndex(hashcode);
+        Node<K,V> firsNodeOnIndex = nodes[index];
+        if (firsNodeOnIndex == null) {
+            return null;
+        }
+        if (firsNodeOnIndex.next == null) {
+            return firsNodeOnIndex.getValue();
+        }
+        Node<K,V> foundNode = getCollisionNode(firsNodeOnIndex, key);
+        return foundNode.getValue();
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    @Override
+    public int hashCode(K key) {
+        int hashcode = Objects.hash(key);
+        return hashcode < 0 ? (hashcode * -1) : hashcode;
+    }
+
+    private static class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(K key, V value) {
+            this.key = key;
+            this.value = value;
+            this.next = null;
+        }
+
+        public K getKey() {
+            return key;
+        }
+
+        public V getValue() {
+            return value;
+        }
+
+        public void setValue(V value) {
+            this.value = value;
+        }
+
+        public void setNext(Node<K, V> next) {
+            this.next = next;
+        }
+
+        @Override
+        public boolean equals(Object key) {
+            return Objects.equals(key, this.key);
+        }
+    }
+
+    private int defineIndex(int hashcode) {
+        return hashcode % capacity;
+    }
+
+    private Node<K,V> getLastCollisionNode(int index) {
+        if (nodes[index] == null) {
+            return null;
+        }
+        Node<K,V> lastNode = nodes[index];
+        while (lastNode.next != null) {
+            lastNode = lastNode.next;
+        }
+        return lastNode;
+    }
+
+    private Node<K,V> getCollisionNode(Node<K,V> nextNode, K key) {
+        if (nextNode.next != null && !nextNode.equals(key)) {
+            nextNode = getCollisionNode(nextNode.next, key);
+        }
+        return nextNode;
+    }
+
+    private void reassignSameKeyValues(Node<K,V> newNode, Node<K,V> currentNode) {
```

Comment:
Try to use simpler vocabulary
```suggestion
    private void putNewNode(Node<K,V> newNode, Node<K,V> currentNode) {
```

Reply:
++, changed